ui/widgets/nuxeo-collapsible.js

```diff
@@ -0,0 +1,171 @@
+   * ----------------|-------------|----------
+   * `--nuxeo-collapsible-heading-padding` | Heading padding | `4px`
+   * `--nuxeo-collapsible-icon-height` | Chevron height | `20px`
+   * `--nuxeo-collapsible-icon-width` | Chevron width | `20px`
+   *
+   * @memberof Nuxeo
+   *
+   */
+
+  class Collapsible extends Nuxeo.Element {
+    static get template() {
+      return html`
+        <style>
+          button {
+            height: 100%;
+            width: 100%;
+            display: flex;
+            justify-content: space-between;
+            align-items: center;
+            margin: 0;
+            padding: var(--nuxeo-collapsible-heading-padding, 4px);
+            background: transparent;
+            border: none;
+            cursor: pointer;
+            text-decoration: none;
+            font-size: 1rem;
+          }
+          .heading {
+            width: calc(100% - var(--nuxeo-collapsible-icon-width, 20px));
+            overflow-x: hidden;
+            text-align: start;
+          }
+          iron-icon {
+            --iron-icon-height: var(--nuxeo-collapsible-icon-height, 20px);
+            --iron-icon-width: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+          .start {
+            flex-direction: row-reverse;
+            justify-content: flex-end;
+          }
+
+          .start-alignment {
+            margin-inline-start: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+          iron-collapse {
+            padding: 0 var(--nuxeo-collapsible-heading-padding, 4px);
+          }
+        </style>
+
+        <button aria-expanded="[[expanded]]" on-tap="_toggle" class$="[[_changeChevronPosition(chevronAtStart)]]">
+          <slot class="heading" name="heading"></slot>
+          <iron-icon icon="[[_toggleIcon(expanded)]]" toggle hidden$="[[!chevron]]"></iron-icon>
+        </button>
+        <iron-collapse
+          opened="[[expanded]]"
+          no-animation="[[!animation]]"
+          class$="[[_alignCollapsedContent(chevronAtStart)]]"
+        >
+          <slot></slot>
+        </iron-collapse>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-collapsible';
+    }
+
+    static get properties() {
+      return {
+        /**
+         * Set animation to false to disable animations
+         */
+        animation: {
+          type: Boolean,
+          value: true,
+        },
+
+        /**
+         * Show or hide the chevron icon
+         */
+        chevron: {
+          type: Boolean,
+          value: false,
+        },
+
+        /**
+         * Chevron icon position. By default it is positioned at the end of the heading,
+         * but it can be swapped to the its beginning
+         */
+        chevronAtStart: {
+          type: Boolean,
+          value: false,
+        },
+
+        /**
+         * Expanded or collapsed by default
+         */
+        expanded: {
+          type: Boolean,
+          value: false,
+          reflectToAttribute: true,
```

Comment:
Is this just for the first rendering, right?
If we set `expanded` and then we start clicking the element it will not keep our "configuration".

Reply:
Yes, this is only meant for the first render. After it you can collapse the element.